contracts/YieldFarm.sol

```diff
@@ -68,14 +68,14 @@ contract YieldFarm {
             epochId = NR_OF_EPOCHS;
         }
 
-        emit MassHarvest(msg.sender, epochId.sub(lastEpochIdHarvested[msg.sender]), totalDistributedValue);
-
         for (uint128 i = lastEpochIdHarvested[msg.sender] + 1; i <= epochId; i++) {
             // i = epochId
             // compute distributed Value and do one single transfer at the end
             totalDistributedValue += _harvest(i);
         }
 
+        emit MassHarvest(msg.sender, epochId.sub(lastEpochIdHarvested[msg.sender]), totalDistributedValue);
```

Comment:
Would it make sense to move it inside the `if (totalDistributedValue > 0)`? Or will we need the 0 mass harvest events also?

Reply:
I was thinking that it makes sense to have it outside that `if` just so we know that a `massHarvest` was executed, regardless of the value distributed. For example, if we end up displaying it in the app, it would be more complete in the context of the user to show all actions.

Reply:
Makes sense 👍